engine/table/src/main/java/io/deephaven/engine/util/PythonScopeJpyImpl.java

```diff
@@ -89,30 +93,106 @@ public Object convertValue(PyObject value) {
+            this.vectorizable = vectorizable;
+        }
+
+        public boolean isVectorizable() {
+            return vectorizable;
+        }
+
+
         public CallableWrapper(PyObject pyObject) {
             this.pyObject = pyObject;
         }
 
+        private void prepareSignature() {
+            if (pyObject.getType().equals(NUMBA_VECTORIZED_FUNC_TYPE)) {
+                List<PyObject> params = pyObject.getAttribute("types").asList();
+                if (params.isEmpty()) {
+                    throw new IllegalArgumentException("numba vectorized function must have an explicit signature.");
+                }
+                // numba allows a vectorized function to have multiple signatures, only the first one
+                // will be accepted by DH
+                signature = params.get(0).getStringValue();
+                vectorized = true;
+            } else if (pyObject.hasAttribute("dh_vectorized")) {
+                signature = pyObject.getAttribute("signature").toString();
+                vectorized = true;
+            } else {
+                signature = dh_table_module.call("_encode_signature", pyObject).toString();
+                vectorized = false;
+            }
+        }
+
+        public void parseSignature() {
+
+            prepareSignature();
+
+            // the 'types' field of a vectorized function takes the form of
+            // '[parameter-type-char*]->[return-type-char]',
+            // eg. [ll->d] defines two int64 (long) arguments and a double return type.
+            if (signature == null || signature.length() == 0) {
+                throw new IllegalStateException("signature should always be available.");
+            }
+
+            char numpyTypeCode = signature.charAt(signature.length() - 1);
+            Class<?> returnType = numpyType2JavaClass.get(numpyTypeCode);
+            if (returnType == null) {
+                throw new IllegalStateException(
+                        "vectorized functions should always have an integral, floating point, boolean, or Object return type.");
+            }
+
+            List<Class<?>> paramTypes = new ArrayList<>();
+            for (char numpyTypeChar : signature.toCharArray()) {
+                if (numpyTypeChar != '-') {
+                    Class<?> paramType = numpyType2JavaClass.get(numpyTypeChar);
+                    if (paramType == null) {
+                        throw new IllegalStateException(
+                                "parameters of vectorized functions should always be of integral, floating point, boolean, or Object type.");
+                    }
+                    paramTypes.add(numpyType2JavaClass.get(numpyTypeChar));
+                } else {
+                    break;
+                }
+            }
+
+            if (paramTypes.size() == 0) {
+                throw new IllegalArgumentException("vectorized functions must have at least one parameter.");
+            }
```

Comment:
I don't understand why this is the case.  It removes useful random functions from vectorization.

Reply:
I was bummed by this too, but there isn't a workable way to communicate to the Python decorator the size of the chunk at the moment. A random/default chunk size didn't work for me as the table size is fixed and chunk copy raises an exception when source and dest aren't of the same size.

Reply:
Worst case, create a ticket and add zero-args to the python todo list.